apps/full-site-editing/full-site-editing-plugin/full-site-editing/class-full-site-editing.php

```diff
@@ -368,6 +363,27 @@ public function merge_template_and_post( $post ) {
 		$post->post_content = preg_replace( '@(<!-- wp:a8c/post-content)(.*?)(/-->)@', "$1$2-->$post->post_content<!-- /wp:a8c/post-content -->", $template_content );
 	}
 
+	/**
+	 * Detects if we are in a context where the template and post should be merged.
+	 *
+	 * Conditions:
+	 * 1. in a REST API request (either flavour)
+	 * 2. OR on a block editor screen (inlined requests using `rest_preload_api_request` )
+	 * 3. AND editing a post_type that supports full site editing
+	 *
+	 * @return bool
+	 */
+	private function should_merge_template_and_post() {
+		$is_rest_api_wpcom = ( defined( 'REST_API_REQUEST' ) && REST_API_REQUEST );
+		$is_rest_api_core = ( defined( 'REST_REQUEST' ) && REST_REQUEST );
```

Comment:
Thanks for the clear names !

Reply:
Ha, there's been a lot of confusion going around with the many REST APIs, thought I'd do my part.